op_crates/fetch/lib.rs

```diff
@@ -155,39 +156,44 @@ where
   }
   //debug!("Before fetch {}", url);
 
-  let res = request.send().await?;
-
-  //debug!("Fetch response {}", url);
-  let status = res.status();
-  let mut res_headers = Vec::new();
-  for (key, val) in res.headers().iter() {
-    let key_string = key.to_string();
-
-    if val.as_bytes().is_ascii() {
-      res_headers.push((key_string, val.to_str().unwrap().to_owned()))
-    } else {
-      res_headers.push((
-        key_string,
-        val
-          .as_bytes()
-          .iter()
-          .map(|&c| c as char)
-          .collect::<String>(),
-      ));
+  if let Ok(res) = request.send().await {
+    //debug!("Fetch response {}", url);
+    let status = res.status();
+    let mut res_headers = Vec::new();
+    for (key, val) in res.headers().iter() {
+      let key_string = key.to_string();
+
+      if val.as_bytes().is_ascii() {
+        res_headers.push((key_string, val.to_str().unwrap().to_owned()))
+      } else {
+        res_headers.push((
+          key_string,
+          val
+            .as_bytes()
+            .iter()
+            .map(|&c| c as char)
+            .collect::<String>(),
+        ));
+      }
     }
+
+    let rid = state.borrow_mut().resource_table.add(HttpBodyResource {
+      response: AsyncRefCell::new(res),
+      cancel: Default::default(),
+    });
+
+    return Ok(json!({
+      "bodyRid": rid,
+      "status": status.as_u16(),
+      "statusText": status.canonical_reason().unwrap_or(""),
+      "headers": res_headers
+    }));
   }
 
-  let rid = state.borrow_mut().resource_table.add(HttpBodyResource {
-    response: AsyncRefCell::new(res),
-    cancel: Default::default(),
-  });
-
-  Ok(json!({
-    "bodyRid": rid,
-    "status": status.as_u16(),
-    "statusText": status.canonical_reason().unwrap_or(""),
-    "headers": res_headers
-  }))
+  return Err(custom_error(
+    "Http",
+    format!("error trying to connect ({})", url),
```

Comment:
The error message defined here is according to that from `fetch_test.ts` line 31. I believe this can be improved for clarity. Any suggestions?

Reply:
I don't think you can get any more specific with reqwest than checking if `err.is_request()` (except maybe scanning the error message for a shibboleth), and that doesn't tell you in any detail what was wrong with the request, so this seems fine to me, it's a step up from the status quo.

Perhaps for troubleshooting purposes print the error with `eprintln!`?

Reply:
It's possible, but I'm still not sure if additional error message would be necessary since this is just fetch API.